internal/app/singularity/pull.go

```diff
@@ -75,23 +76,42 @@ func LibraryPull(name, ref, transport, fullURI, libraryURI, keyServerURL, authTo
 		return fmt.Errorf("failed getting image from the library")
 	}
 
-	imagePath := cache.LibraryImage(libraryImage.Hash, imageName)
-	exists, err := cache.LibraryImageExists(libraryImage.Hash, imageName)
-	if err != nil {
-		return fmt.Errorf("unable to check if %s exists: %v", imagePath, err)
-	}
-	if !exists {
-		sylog.Infof("Downloading library image")
+	imagePath := ""
+
+	if noCache {
+		// Dont use cached image
+		// TODO: Remove file.Name() when done with it
```

Comment:
Here too

Reply:
Yup, still a TODO, I will look into it a bit more...

Reply:
Not a TODO anymore :smile: